hack/tools/go.mod

```diff
@@ -1,16 +1,17 @@
 module sigs.k8s.io/cluster-api-provider-ibmcloud/hack/tools
 
 go 1.22.0
+toolchain go1.22.5
```

Comment:
Created an issue to avoid this - https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/issues/2048